src/openrct2/world/footpath.c

```diff
@@ -1773,15 +1786,14 @@ sint32 footpath_is_connected_to_map_edge(sint32 x, sint32 y, sint32 z, sint32 di
 
 bool footpath_element_is_sloped(const rct_tile_element * tileElement)
 {
-    return tileElement->properties.path.type & FOOTPATH_PROPERTIES_FLAG_IS_SLOPED;
+    return (tileElement->properties.path.type & FOOTPATH_PROPERTIES_FLAG_IS_SLOPED) != 0;
 }
 
 void footpath_element_set_sloped(rct_tile_element * tileElement, bool isSloped)
 {
+    tileElement->properties.path.type &= ~FOOTPATH_PROPERTIES_FLAG_IS_SLOPED;
```

Comment:
Why are you doing this? It's an unnecessary operation if the footpath is sloped.